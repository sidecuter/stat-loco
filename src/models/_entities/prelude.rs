//! `SeaORM` Entity, @generated by sea-orm-codegen 1.1.10

pub use super::sites::Entity as Sites;
pub use super::user_ids::Entity as UserIds;
pub use super::users::Entity as Users;
