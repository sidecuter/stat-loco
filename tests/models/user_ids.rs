use insta::assert_debug_snapshot;
use loco_rs::testing::prelude::*;
use serial_test::serial;
use stat_api::{app::App, models::user_ids::Model};

macro_rules! configure_insta {
    ($($expr:expr),*) => {
        let mut settings = insta::Settings::clone_current();
        settings.set_prepend_module_to_snapshot(false);
        settings.set_snapshot_suffix("user_ids");
        let _guard = settings.bind_to_scope();
    };
}

#[tokio::test]
#[serial]
async fn can_create() {
    configure_insta!();

    let boot = boot_test::<App>()
        .await
        .expect("Failed to boot test application");

    let res = Model::create_new(&boot.app_context.db).await;

    insta::with_settings!({
        filters => vec![
            (
                r"([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})",
                "PID",
            ),
            (r"id: \d+,", "id: ID,"),
            (
                r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?\+\d{2}:\d{2}",
                "DATE",
            ), // with tz
            (r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+", "DATE"),
            (r"(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})", "DATE"),
        ]
    }, {
        assert_debug_snapshot!(res);
    });
}
