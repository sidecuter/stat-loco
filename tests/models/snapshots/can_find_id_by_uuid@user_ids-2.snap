---
source: tests/models/user_ids.rs
expression: non_existing_user_id_results
---
Err(
    EntityNotFound,
)
